typescript/ws-protocol/src/types.ts

```diff
@@ -40,13 +40,27 @@ export type Channel = {
   schema: string;
   schemaEncoding?: string;
 };
-export type Service = {
+
+type ServiceRequest = {
+  encoding: string;
+  schemaName: string;
+  schemaEncoding: string;
+  schema: string;
+};
+type ServiceRequestDef =
+  | { requestSchema: string; request?: ServiceRequest }
+  | { requestSchema?: string; request: ServiceRequest };
```

Comment:
I'm a bit confused by these types – is it allowed to specify both request & requestSchema? It seems like the types would allow that

Reply:
It's allowed, but only `request` would be used then. Better to make it exclusive?